examples/systemcalls/systemcalls.c

```diff
@@ -92,7 +179,65 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   The rest of the behaviour is same as do_exec()
  *
 */
-
+    int status = -1;
```

Comment:
@chchbuff think about how you could avoid duplicating logic between these two functions.